### PR TITLE
Double-check that the cache directory does not exist

### PR DESCRIPTION
**Pull request by AChep**
Check https://github.com/google/ExoPlayer/issues/8253#issuecomment-760218637 for more info

### CODE REVIEW

**Comment by lcf87** [comment]
Please move the pull request to dev-v2.

**Comment by AChep** [comment]
Will do it this evening.

File: library/core/src/main/java/com/google/android/exoplayer2/upstream/cache/SimpleCache.java [hunk 1]
```diff
@@ -837,7 +837,7 @@ private static long parseUid(String fileName) {
   }
 
   private static void createCacheDirectories(File cacheDir) throws CacheException {
-    if (!cacheDir.mkdirs()) {
+    if (!cacheDir.mkdirs() && !cacheDir.exists()) {
```
Comment (ojw28) [inline library/core/src/main/java/com/google/android/exoplayer2/upstream/cache/SimpleCache.java:840]: Would it be a bit better to use:

`if (!cacheDir.mkdirs() && (!cacheDir.exists() || !cacheDir.isDirectory())) {`

**Comment by ojw28** [comment]
@AChep - Thanks for this. I'll close this PR because I'm pretty certain you'll need to use a fresh one to target `dev-v2` as required. Please send the one for `dev-v2` when you have a chance. I also put one comment inline (see above). Thanks!

**Comment by AChep** [comment]
@ojw28 do you mind if I'll add _more messages and stuff_ if the file is not a directory?

**Pull request closed without merging**